NXDOC-3002: documentation for bulk migration in nuxeo PDF/A conversion 2025 version (#2803)
nuxeo/doc.nuxeo.com-platform-spaces: src/nxdoc/nuxeo-add-ons/nuxeo-pdfa-conversion.md

PDF/A is a version of PDF intended for long-term preservation. It is designed to keep documents self-contained so they can be opened and viewed more consistently over time.
{{/callout}}

## Bulk Migration for Pre-existing Documents

Documents uploaded before the PDF/A package was installed do not have a PDF/A rendition. You can convert them in bulk using the management REST API.

### Triggering a Bulk Migration

Send a POST request to the migration endpoint. Only administrators can perform this operation.

```bash
curl -u Administrator:Administrator -X POST \
  "http://localhost:8080/nuxeo/api/v1/management/pdfa/migrate?query=SELECT * FROM Document WHERE file:content/name IS NOT NULL AND pdfa:content/name IS NULL AND ecm:isVersion = 0 AND ecm:isProxy = 0 AND ecm:isTrashed = 0"
```

The response returns a `commandId` that you can use to track progress.

### Checking Migration Status

Poll the bulk command status until the state is `COMPLETED`:

```bash
curl -u Administrator:Administrator \
  "http://localhost:8080/nuxeo/api/v1/management/bulk/<commandId>"
```

> Once complete, the affected documents will display the **View PDF/A** button and the **Export > PDF/A** option in the Web UI.

Reviewer: ```suggestion
Once complete, the affected documents display the **View PDF/A** button and the **Export > PDF/A** option in the Web UI.
```

### Scoping Migration to a Specific Folder

> To limit migration to documents within a particular folder, adjust the query parameter:

Reviewer: ```suggestion
To limit migration of documents to a particular folder, adjust the query parameter:
```

```bash
curl -u Administrator:Administrator -X POST \
  "http://localhost:8080/nuxeo/api/v1/management/pdfa/migrate?query=SELECT * FROM Document WHERE ecm:path STARTSWITH '/default-domain/workspaces/MyFolder' AND file:content/name IS NOT NULL AND ecm:isVersion = 0"
```

### Behavior Notes

* **Idempotent:** Re-running the migration skips documents that already have a PDF/A rendition. No duplicates are created.

> * **Unsupported files are skipped:** Files with unsupported MIME types (e.g., ZIP archives) are skipped without raising errors.

Reviewer: ```suggestion
* **Unsupported files are skipped:** Files with unsupported MIME types (for example, ZIP archives) are skipped without raising errors.
```
